src/features/utils/sqlfluff.ts

```diff
@@ -0,0 +1,129 @@
+import * as vscode from "vscode";
+import * as path from "path";
+import * as childProcess from "child_process";
+
+import { Configuration } from "../helpers/configuration"
+import { LineDecoder } from "./lineDecoder";
+
+export enum SQLFluffCommand {
+  LINT = 'lint',
+  FIX = 'fix',
+}
+
+export interface SQLFluffCommandOutput {
+  succeeded: boolean;
+  lines: string[];
+}
+
+export interface SQLFluffCommandOptions {
+  targetFileFullPath?: string;
+  fileContents?: string;
+}
+
+export class SQLFluff {
+  public executableNotFound: boolean;
+  public outputChannel: vscode.OutputChannel;
+  private process: childProcess.ChildProcess;
+
+  constructor(outputChannel: vscode.OutputChannel) {
+    this.outputChannel = outputChannel;
+  }
+
+  run(cwd: string, command: SQLFluffCommand, args: string[], options: SQLFluffCommandOptions): Promise<SQLFluffCommandOutput> {
+    if (!options.fileContents && !options.targetFileFullPath) {
+      throw new Error('You must supply either a target file path or the file contents to scan');
+    }
+    if (this.process) {
+      this.process.kill('SIGKILL');
+      this.process = undefined;
+    }
+    const workspaceFolderRootPath = path.normalize(vscode.workspace.workspaceFolders[0].uri.fsPath);
```

Comment:
Looks like we normalize the paths differently in the **lint.ts** and **format.ts** files. Could we change it, either here or in those files, so that we have consistent path normalization?